js/interactive-guides/circuit-breaker/circuit-breaker-callback.js

```diff
@@ -313,34 +313,59 @@ var circuitBreakerCallBack = (function() {
     };
 
     var __listenToEditorForCircuitBreakerAnnotationChanges = function(editor){
-
-        var __showCircuitBreakerInPod = function(){
+        var __listenToContentChanges = function(editorInstance, changes) {
             // Get pod from contentManager
-            var cb = contentManager.getCircuitBreaker(editor.getStepName());
-
+            var cb = contentManager.getCircuitBreaker(editor.getStepName());            
             // Get the parameters from the editor and send to the circuitBreaker
             var content = editor.getEditorContent();
             try{
-            var annotation = content.match(/@CircuitBreaker(.|\n)*\)/g)[0];
-              annotation = annotation.substring(0,annotation.indexOf("public")).trim(); // Get rid of the public Service...
-              var params = annotation.substring(16,annotation.length-1);
-              // params = params.replace('\n','');
-              params = params.replace(/\s/g, ''); // Remove whitespace
-              params = params.split(',');
-              params.forEach(function(element, index){
-                params[index] = element.trim().substring(element.indexOf('=')+1);
-              });
-
-              cb.updateParameters.apply(cb, params);
+                var matchPattern = "public class BankService\\s*{\\s*@CircuitBreaker\\s*(((\\s|\\S)?))*public Service checkBalance";
+                var regexToMatch = new RegExp(matchPattern, "g");
+                var groups = regexToMatch.exec(content);
+                var match = groups[0];
```

Comment:
Sorry, I'm confused as you should be able to use groups[1] to get the parameter value and not have to do another content.match and substring to get out the parameter values.

Reply:
Fixed